### ISSUE TITLE
format_timedelta fails on es_US / narrow

### ISSUE THREAD

**Issue by ikonst** [python-babel/babel#798]
This code:
```python
from datetime import timedelta

from babel.dates import format_timedelta

format_timedelta(timedelta(), format='narrow', locale='es_US')
```
works on babel 2.7 but breaks since babel 2.8 with:
```
  ...
  File ".../site-packages/babel/dates.py", line 944, in format_timedelta
    pattern = patterns[plural_form]
  File ".../site-packages/babel/localedata.py", line 218, in __getitem__
    orig = val = self._data[key]
KeyError: 'other'
```

(I suspect the CLDR 35 to 36 upgrade.)

**Comment by ikonst** [comment]
Thank you!

**Comment by jun66j5** [comment]
Hi @ikonst,

The issue doesn't occur with Babel 2.10.3. I guess that's an installation issue on your environment. Could you please recheck Babel's version?

```pycon
$ pip install --no-cache-dir Babel
Collecting Babel
  Downloading Babel-2.10.3-py3-none-any.whl (9.5 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 9.5/9.5 MB 7.5 MB/s eta 0:00:00
Collecting pytz>=2015.7
  Downloading pytz-2022.1-py2.py3-none-any.whl (503 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 503.5/503.5 kB 22.3 MB/s eta 0:00:00
Installing collected packages: pytz, Babel
Successfully installed Babel-2.10.3 pytz-2022.1
$ python
Python 3.10.4 (main, Apr  8 2022, 17:35:13) [GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import babel
>>> babel.__version__
'2.10.3'
>>> from datetime import timedelta
>>> from babel import dates
>>> dates.format_timedelta(timedelta(minutes=1), format='short', locale='pt_BR')
'1 min'
```

**Comment by ikonst** [comment]
@jun66j5 That's precisely what I was saying, that babel 2.10 has fixed this issue too (which was present with 2.9).

**Comment by jun66j5** [comment]
Okay. Sorry about my wrong comment.
